tests/Catch2_Tests/comparators/ComparatorDisposerTest.cpp

```diff
@@ -0,0 +1,56 @@
+#include "catch2/catch.hpp"
+#include "ApprovalTests/comparators/ComparatorDisposer.h"
+#include "Approvals.h"
+#include <fstream>
+#include <iterator>
+#include <memory>
+
+using namespace ApprovalTests;
+
+namespace
+{
+    class TestComparator : public ApprovalComparator
+    {
+    public:
+        bool contentsAreEquivalent(std::string receivedPath,
+                                   std::string approvedPath) const override
+        {
+            std::ifstream receivedStream(receivedPath.c_str());
+            std::ifstream approvedStream(approvedPath.c_str());
+
+            using InputIter = std::istreambuf_iterator<char>;
+
+            auto receivedText =
+                std::string(InputIter{receivedStream}, InputIter{});
+            auto approvedText =
+                std::string(InputIter{approvedStream}, InputIter{});
+
+            return (receivedText == "foo\n") && (approvedText == "bar\n");
+        }
+    };
+}
+
+TEST_CASE("Type traits of ComparatorDisposer")
+{
+
+    REQUIRE_FALSE(std::is_copy_constructible<ComparatorDisposer>::value);
+    REQUIRE_FALSE(std::is_copy_assignable<ComparatorDisposer>::value);
+
+    REQUIRE(std::is_move_constructible<ComparatorDisposer>::value);
+    REQUIRE_FALSE(std::is_move_assignable<ComparatorDisposer>::value);
+}
+
+TEST_CASE("ComparatorDisposer object can be moved without disposing a "
+          "comparator twice")
+{
+    std::unique_ptr<ComparatorDisposer> disposer;
+
+    {
+        auto temporaryDisposer = FileApprover::registerComparatorForExtension(
+            ".txt", std::make_shared<TestComparator>());
+
+        disposer.reset(new ComparatorDisposer(std::move(temporaryDisposer)));
```

Comment:
It's really helpful to see how you guaranteed that a move operation would happen, for the testing. 

An explanatory comment would be really helpful, to explain the purpose of lines 46, 49 and 52.

Reply:
I've added comments before each step in the test.